File: week-10/yc_13913_hide_and_seek_4.java
import java.io.*;
import java.util.*;

public class Main {
    static final int MAX = 100000;
    static int[] dist = new int[MAX + 1];
    static int[] prev = new int[MAX + 1];

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        Arrays.fill(dist, -1);
        Arrays.fill(prev, -1);

        bfs(N, K);

        List<Integer> path = new ArrayList<>();
        for (int cur = K; cur != -1; cur = prev[cur]) {
            path.add(cur);
        }
        Collections.reverse(path);

        StringBuilder sb = new StringBuilder();
        sb.append(dist[K]).append("\n");
        for (int p : path) sb.append(p).append(" ");
        System.out.println(sb);
    }

    static void bfs(int N, int K) {
        Queue<Integer> q = new LinkedList<>();
        q.offer(N);
        dist[N] = 0;

        while (!q.isEmpty()) {
            int x = q.poll();

            if (x == K) return;

            int[] nexts = {x - 1, x + 1, x * 2};

            for (int nx : nexts) {
                if (nx >= 0 && nx <= MAX && dist[nx] == -1) {
                    dist[nx] = dist[x] + 1;
                    prev[nx] = x;
                    q.offer(nx);
                }
            }
        }
    }
}

File: week-10/yc_13913_hide_and_seek_4.cs
using System;
using System.Collections.Generic;

class Program
{
    const int MAX = 100000;
    static int[] dist = new int[MAX + 1];
    static int[] prev = new int[MAX + 1];

    static void Main()
    {
        string[] input = Console.ReadLine().Split();
        int N = int.Parse(input[0]);
        int K = int.Parse(input[1]);

        Array.Fill(dist, -1);
        Array.Fill(prev, -1);

        BFS(N, K);

        List<int> path = new List<int>();
        for (int cur = K; cur != -1; cur = prev[cur])
            path.Add(cur);

        path.Reverse();

        Console.WriteLine(dist[K]);
        Console.WriteLine(string.Join(" ", path));
    }

    static void BFS(int N, int K)
    {
        Queue<int> q = new Queue<int>();
        q.Enqueue(N);
        dist[N] = 0;

        while (q.Count > 0)
        {
            int x = q.Dequeue();

            if (x == K) return;

            int[] nexts = { x - 1, x + 1, x * 2 };

            foreach (int nx in nexts)
            {
                if (nx >= 0 && nx <= MAX && dist[nx] == -1)
                {
                    dist[nx] = dist[x] + 1;
                    prev[nx] = x;
                    q.Enqueue(nx);
                }
            }
        }
    }
}
